dfc/tests/regression_test.go

```diff
@@ -92,6 +92,119 @@ var (
 	failLRU = ""
 )
 
+// Note: This test is most effective when multiple targets are deployed.
+func TestLocalListBucketGetTargetURL(t *testing.T) {
+	const (
+		num      = 1000
+		filesize = uint64(1024)
+		seed     = int64(111)
+		bucket   = TestLocalBucketName
+	)
+	var (
+		filenameCh = make(chan string, num)
+		errch      = make(chan error, 100)
+		sgl        *dfc.SGLIO
+		targets    = make(map[string]struct{})
+	)
+
+	if usingSG {
+		sgl = dfc.NewSGLIO(filesize)
+		defer sgl.Free()
+	}
+
+	err := client.CreateLocalBucket(proxyurl, bucket)
+	checkFatal(err, t)
+
+	defer func() {
+		err = client.DestroyLocalBucket(proxyurl, bucket)
+		checkFatal(err, t)
+	}()
+
+	putRandomFiles(0, seed, filesize, num, bucket, t, nil, errch, filenameCh, SmokeDir, SmokeStr, "", true, sgl)
+	selectErr(errch, "put", t, false)
+
+	msg := &dfc.GetMsg{GetPageSize: int(pagesize), GetTargetURL: true}
+	bl, err := client.ListBucket(proxyurl, bucket, msg, num)
+	checkFatal(err, t)
+
+	if len(bl.Entries) != num {
+		t.Errorf("Expected %d bucket list entries, found %d\n", num, len(bl.Entries))
+	}
+
+	for _, e := range bl.Entries {
+		if e.TargetURL == "" {
+			t.Error("Target URL in response is empty")
+		}
+		if _, ok := targets[e.TargetURL]; !ok {
+			targets[e.TargetURL] = struct{}{}
+		}
+		l, _, err := client.Get(e.TargetURL, bucket, e.Name, nil, nil, false, false)
+		checkFatal(err, t)
+		if uint64(l) != filesize {
+			t.Errorf("Expected filesize: %d, actual filesize: %d\n", filesize, l)
+		}
+	}
+
+	smap, err := client.GetClusterMap(proxyurl)
+	checkFatal(err, t)
+	if len(smap.Tmap) != len(targets) { // The objects should have been distributed to all targets
+		t.Errorf("Expected %d different target URLs, actual: %d different target URLs", len(smap.Tmap), len(targets))
+	}
+}
+
+// Note: This test is most effective when multiple targets are deployed.
+func TestCloudListBucketGetTargetURL(t *testing.T) {
+	const (
+		num      = 20
```

Comment:
It is again about the last check in the function. If you run many targets it is probable that your 20 objects will be on less targets than the test expects.
Maybe make check less strict? E.g, check that the number of different targets in response in greater than 1(for case number of targets > 1) or 1(for 1 target)?